dockstore-webservice/src/main/java/io/openapi/api/impl/ToolsApiServiceImpl.java

```diff
@@ -288,24 +289,41 @@ public Response toolsGet(String id, String alias, String toolClass, String descr
         } else if (alias != null) {
             all.add(toolDAO.getGenericEntryByAlias(alias));
         } else {
-            if (toolClass == null || COMMAND_LINE_TOOL.equalsIgnoreCase(toolClass)) {
+            // If there is no matching descriptor language, then do not add any tools or workflows.
+            DescriptorLanguage descriptorLanguage = null;
+            boolean descriptorLanguageFound = true;
+
+            // Tricky case for GALAXY because it doesn't match the rules of the other languages
+            if ("galaxy".equalsIgnoreCase(descriptorType)) {
```

Comment:
This could go inside the if on line 301, where you know `descriptorType` is not null. More from a readability perspective